console/views.py

```diff
@@ -65,3 +71,53 @@ def decrypt():
 
     else:
         return render_template('decrypt.html')
+
+
+def get_all_responses():
+    data = SurveyResponse.query.all()
+    return data
+
+
+def get_store_responses(tx_id, ru_ref, survey_id, datetime_earliest, datetime_latest):
+    responses = get_all_responses()
+    store_data = []
+    for response in responses:
+        store_data.append(response.data)
+
+    filtered_data = []
+    for data in store_data:
+        if (tx_id == '') or (data['tx_id'] == tx_id):
+            if (ru_ref == '') or (data['metadata']['ru_ref'] == ru_ref):
+                if (survey_id == '') or (data['survey_id'] == survey_id):
+                    filtered_data.append(data)
+
+    datetime_earliest_f = datetime.strptime(datetime_earliest, "%Y-%m-%dT%H:%M")
+    datetime_latest_f = datetime.strptime(datetime_latest, "%Y-%m-%dT%H:%M")
+
+    final_data = []
+    for data in filtered_data:
+        data_datetime_string = data['submitted_at'][:19]
+        data_datetime = datetime.strptime(data_datetime_string, "%Y-%m-%dT%H:%M:%S")
+        if (data_datetime > datetime_earliest_f) and (data_datetime < datetime_latest_f):
+            final_data.append(data)
+
+    return final_data
+
+
+@app.route('/store', methods=['GET', 'POST'])
+@flask_security.roles_required('SDX-Developer')
+def store():
+    if request.method == 'GET':
+        tx_id = request.args.get('tx_id', type=str, default='')
+        ru_ref = request.args.get('ru_ref', type=str, default='')
+        survey_id = request.args.get('survey_id', type=str, default='')
+        datetime_earliest = request.args.get('datetime_earliest', type=str, default='1990-01-01T00:00')
+        datetime_latest = request.args.get('datetime_latest', type=str, default='2020-01-01T00:00')
+
+        store_data = get_store_responses(tx_id, ru_ref, survey_id, datetime_earliest, datetime_latest)
+
+        return render_template('store.html', data=store_data)
+
+    else:
+        data = request.data
+        send_data(settings.SDX_STORE_URL + "responses", data=data, request_type="POST")
```

Comment:
```python
else:
    send_data(settings.SDX_STORE_URL + "responses", data=request.data, request_type="POST")
```

We don't use the data var again so don't worry about assigning it, instead access the request data directly